07.ls_object/output.rb

```diff
@@ -3,69 +3,68 @@
 class Output
   def initialize(options)
     @options = options
-    paths = if @options.include?(:a)
-              Dir.entries('.').sort
-            else
-              Dir.glob('*')
-            end
-
-    @contents = if @options.include?(:r)
-                  paths.map { |path| Content.new(path) }.reverse
-                else
-                  paths.map { |path| Content.new(path) }
-                end
+    paths = @options.include?(:a) ? Dir.entries('.').sort : Dir.glob('*')
+    contents = paths.map { |path| Content.new(path) } # @contentsと名前が被るのですが、いい案が思いつかなかったです😓
+    @contents = @options.include?(:r) ? contents.reverse : contents
   end
 
   def display
     if @options.include?(:l)
-      max_blocks = @contents.sum(&:blocks)
-      rjust_blank_sizes = calcurate_add_blank_sizes # 良い変数名が思いつかず、しっくりきていないです。。
-      output_detail_lines(max_blocks, rjust_blank_sizes)
+      total_block_size = @contents.sum(&:blocks)
+      output_detail_lines(total_block_size)
     else
-      max_path = @contents.map { |content| content.path.length }.max
-      lines = @contents.map { |content| content.path.ljust(max_path) }
-      output_three_column_format(lines)
+      pathname_widths = @contents.map { |content| content.path.length }.max
+      lines = @contents.map { |content| content.path.ljust(pathname_widths) }
+      output_pathname(lines)
     end
   end
 
   private
 
-  def calcurate_add_blank_sizes
-    max_hardlink = @contents.map { |content| content.hardlink.to_s.size }.max
-    max_owner_name = @contents.map { |content| content.owner_name.length }.max
-    max_group_name = @contents.map { |content| content.group_name.length }.max
-    max_bytesize = @contents.map { |content| content.bytesize.to_s.size }.max
+  def output_detail_lines(total_block_size)
+    max_hardlink = find_max_hardlink
```

Comment:
```suggestion
    max_hardlink = @contents.map { |content| content.hardlink.to_s.size }.max
```

メソッド化しない、というのも一手です。

https://bootcamp.fjord.jp/pages/when-you-should-define-method